File: 285-twopointerReverseString.py
def reverseOnlyLetters(s):
    a=list(s)
    l,r=0,len(s)-1
    
    while l<r:
        while l<r and not a[l].isalpha():
            l+=1
        while l<r and not a[r].isalpha():
            r-=1
        a[l],a[r]=a[r],a[l]
        l+=1
        r-=1
    return "".join(a)


s="a_b_cde"
print(reverseOnlyLetters(s))

File: 285-twopointerReverseString.cpp
#include<string>
#include<iostream>

using DT=std::string;

void reverseStringForEnglishLetters(DT& s)
{
    if(s.empty())
        return;
    
    int R=s.size()-1,L=0;
    while(L<R)
    {
        while(L<R && !std::isalpha(s[L]))
            L+=1;
        while(L<R && !std::isalpha(s[R]))
            R-=1;
        
        char t=s[L];
        s[L]=s[R];
        s[R]=t;

        L+=1;
        R-=1;    
    }
}

int main()
{
   DT s="a_b_cde";
   reverseStringForEnglishLetters(s);
   std::cout<<s;
   return 0;
}